corehq/apps/app_manager/management/commands/audit_case_search_config.py

```diff
@@ -0,0 +1,59 @@
+from collections import namedtuple
+
+from django.core.management.base import BaseCommand
+
+from corehq.apps.app_manager.models import Application
+from corehq.apps.app_manager.management.commands.helpers import get_all_app_ids
+from corehq.toggles import SYNC_SEARCH_CASE_CLAIM
+
+PropertyInfo = namedtuple("PropertyInfo", "domain app_id version module_unique_id name")
+
+
+class Command(BaseCommand):
+    help = ("Pull all case search properties that allow blank values from all case search apps")
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '-i',
+            '--include-builds',
+            action='store_true',
+            dest='include_builds',
+            help='Include saved builds, not just current apps',
+        )
+        parser.add_argument(
+            '-d',
+            '--domain',
+            action='store',
+            help='Audit a single domain',
+        )
+
+    def handle(self, **options):
+        include_builds = options['include_builds']
+        if options['domain']:
+            domains = [options['domain']]
+        else:
+            domains = sorted(SYNC_SEARCH_CASE_CLAIM.get_enabled_domains())
+
+        results = []
+        domain_count = 0
+        for domain in domains:
+            domain_count += 1
+            app_ids = get_all_app_ids(domain, include_builds=include_builds)
+            print(f"Looking at {domain} ({domain_count} of {len(domains)} domains), which has {len(app_ids)} apps")
+            for app_id in app_ids:
+                doc = Application.get_db().get(app_id)
+                for index, module in enumerate(doc.get("modules", [])):
+                    if module.get("search_config", {}):
+                        for prop in module.get("search_config", {}).get("properties", []):
+                            if prop.get("allow_blank_value", False):
```

Comment:
Guessing the idea is that this is a jumping-off point to edit the script to check for other things?

Reply:
yes